content/blog/2016/2016-05-19-update-plugin-for-pipeline-tutorial.md

```diff
@@ -0,0 +1,189 @@
+
+After a few days of rando n00b mails to the mailing lists, asking dumb questions in IRC, absorbing bits of knowledge from READMEs in various github repos, and just poking at source code, I eventually got it all sorted out.  So, in hopes of saving the next person a little time, and encouraging plugin authors to invest the time to get their plugins working with Pipeline, here are some notes about what I learned.
+
+Spoiler: if you're just interested in looking at the individual git commits that I made on may way to getting the plugin working with Pipeline, have a look at [this github branch](https://github.com/cprice404/gatling-plugin/commits/feature/master/compatibility-with-jenkins-pipeline.individual-commits).
+
+### Creating a pipeline step
+
+The main task that the Gatling plugin performs is to archive Gatling reports after a run.  I figured that the end game for this exercise was that I was going to end up with a Pipeline "step" that I could include in my Pipeline scripts, to trigger the archiving of the reports.  So my first thought was to look for an existing plugin / Pipeline "step" that was doing something roughly similar, so that I could use it as a model.  The Pipeline "Snippet Generator" feature (create a pipeline job, scroll down to the "Definition" section of its configuration, and check the "Snippet Generator" checkbox) is really helpful for figuring out stuff like this; it is automatically populated with all of the steps that are valid on your server (based on which plugins you have installed), so you can use it to verify whether or not your custom "step" is recognized, and also to look at examples of existing steps.
+
+Looking through the list of existing steps, I figured that the `archive` step was pretty likely to be similar to what I needed for the gatling plugin:
+
+![archive snippet](images/2016-05-19-update-plugin-for-pipeline-tutorial/05_snippet_generator_archive.png)
+
+So, I started poking around to see what magic it was that made that `archive` step show up there.  There are some mentions of this in the [pipeline-plugin DEVGUIDE.md](https://github.com/jenkinsci/pipeline-plugin/blob/6cffbecd874b924677ce3b3c5b1e0e2f45689cc5/DEVGUIDE.md#build-steps) and the [workflow-step-api-plugin README.md](https://github.com/jenkinsci/workflow-step-api-plugin/blob/ee8f181c5561d70207a6b84b4d91ca24312c8a39/README.md), but the real breakthrough for me was finding the [definition of the `archive` step in the `workflow-basic-steps-plugin` source code](https://github.com/jenkinsci/workflow-basic-steps-plugin/blob/300fe6c02b41f072e50a501cfec3e2f425048446/src/main/java/org/jenkinsci/plugins/workflow/steps/ArtifactArchiverStep.java#L37-L53).
```

Comment:
Yep, have had issues in the past where linking to a live branch means the line numbers can get out of whack over time, so that's why I linked to individual commits.